ext/url/01_urlpattern.js

```diff
@@ -137,33 +137,65 @@ class SampledLRUCache {
 
 const matchInputCache = new SampledLRUCache(4096);
 
+const _hasRegExpGroups = Symbol("[[hasRegExpGroups]]");
+
 class URLPattern {
   /** @type {Components} */
   [_components];
+  [_hasRegExpGroups];
 
   #reusedResult;
 
   /**
    * @param {URLPatternInput} input
-   * @param {string} [baseURL]
+   * @param {string} [baseURLOrOptions]
+   * @param {string} [maybeOptions]
    */
-  constructor(input, baseURL = undefined) {
+  constructor(
+    input,
+    baseURLOrOptions = { __proto__: null },
```

Comment:
I mean this is wrong. Please apply the suggestions I made on slack where you default to undefined.